src/quick-lint-js/parse.h

```diff
@@ -2515,6 +2515,13 @@ class parser {
 
     if (this->peek().type == token_type::kw_else) {
       this->skip();
+      if (this->peek().type == token_type::semicolon) {
+        source_code_span semicolon = this->peek().span();
```

Comment:
Is there somewhere to move this out (`this->warn_on_redundant_semicolon();` ). I don't like the inner if.

Reply:
> Is there somewhere to move this out (`this->warn_on_redundant_semicolon();` ). I don't like the inner if.

I understand your sentiment. You can certainly make another member function in `parser`.

Normally I only make functions in this class to avoid code duplication or for major syntax elements. Therefore, what you wrote is typical style.